src/ddmd/target.d

```diff
@@ -284,6 +285,82 @@ struct Target
         return 0;
     }
 
+    /**
+     * Checks whether the target supports operation `op` for vectors of type `type`.
+     * For binary ops `t2` is the type of the 2nd operand.
+     *
+     * Returns:
+     *      true if the operation is not supported.
```

Comment:
Yes, this is annoying, but ddmd.expression is full of check functions that use exactly this logic.
I'd be in favor of replacing all of those, but I'm not going to introduce a single random function that is different from the rest without a particular reason.